### DISCUSSION TITLE
Restore Terminals feature

### DISCUSSION THREAD

**Discussion by drmrbrewer** [coder/code-server#4552] [Q&A]
Given that extensions from the Marketplace are not available in code-server (see [here](https://github.com/cdr/code-server/blob/main/docs/FAQ.md#why-cant-code-server-use-microsofts-extension-marketplace)), is there a "native" way to implement what [this extension](https://marketplace.visualstudio.com/items?itemName=EthanSK.restore-terminals) does?  

I'd find it really useful to be able to restore all my terminals, particularly since I'm often having to restart the code-server host because of an issue like that [described here](https://github.com/microsoft/vscode/issues/128263), where I get a dialog pop up for simple file operations like add, rename, move... not sure what's causing that... I can dismiss the dialog immediately and the file operation proceeds without any apparent issue.  However, I also find that version control isn't working properly in this state (auto detection of git history / changes etc)... so a restart is the only thing that seems to fix this... but I lose all my terminals in doing so.

**Comment by jsjoeio** [comment]
> is there a "native" way to implement what this extension does?

What if the author published the extension on Open VSX? Then you could use it in code-server. Would that work?

**Reply by drmrbrewer** [reply]
So the extensions are in the exact same format in both places, i.e. no extra effort for the author (apart from pushing to two places rather than one)?

**Reply by jsjoeio** [reply]
> no extra effort for the author (apart from pushing to two places rather than one)

Exactly! I haven't published one on Open VSX myself but it shouldn't be too difficult. [This](https://github.com/open-vsx/publish-extensions) might help.
